flang/lib/Lower/OpenMP.cpp

```diff
@@ -178,14 +179,27 @@ genOMP(Fortran::lower::AbstractConverter &converter,
                          &clause.u)) {
         const Fortran::parser::OmpObjectList &ompObjectList = copyinClause->v;
         genObjectList(ompObjectList, converter, copyinClauseOperands);
-      }
+      } else if(const auto &allocateClause = 
+                  std::get_if<Fortran::parser::OmpAllocateClause>(&clause.u)){
+        if (const auto &allocatorValue = 
+              std::get<std::optional<Fortran::parser::OmpAllocateClause::Allocator>>(allocateClause->t)){
+          allocator = fir::getBase(converter.genExprValue(*Fortran::semantics::GetExpr(allocatorValue->v)));
+          allocatorOperands.push_back(allocator);
+        } else{
```

Comment:
Not sure of this piece of code? Could you please elaborate. FWIW you're pushing into the vector ? but again as my previous comment ^above. Do we need a vector in first place.

Reply:
Here I am checking whether the allocate clause has an allocator specified, in which case I push it into the vector. If it does not have an allocator specified, then I push a default allocator into the vector.